### DISCUSSION TITLE
RTranslator 2.0.2

### DISCUSSION THREAD

**Discussion by niedev** [niedev/RTranslator#41] [General]
Added the ability to use any TTS (thanks to @JingziC).

Now languages ​​not supported by the current TTS will still be usable but without TTS (thanks to @JingziC).

Fixed a bug that made TTS not work on some phones (thanks to @starlightjy).

<hr /><em>This discussion was created from the release <a href='https://github.com/niedev/RTranslator/releases/tag/2.0.2'>RTranslator 2.0.2</a>.</em>

**Comment by jark006** [comment]
TTS work now, thanks

**Comment by oldsento** [comment]
[BUG]The Android phone Honor X10, the Android 10 system,and Software version:2.0.2, and the translation of the long text have been rendered mechanically incorrectly: the following phrases have been repeated several times:

The original text:
喜悦中。我终于也可以细细品味这份愉悦，都不知道究竟忍耐了多久。

    先天下之忧而忧，后天下之乐而乐。

    马基维利曾说过，人的脑袋分成三种。第一种，很了解自我能力；第二种，能理解他人的了解；第三种，既不了解自己的能力，也不了解他人的力量。第一种极其优秀，第二种也相当优异，至于第三种则相当无能──无须多言，我是第一种脑袋。

    其证据就是，我十分轻易地置众人于下。哼哼！

    第一次定期测验个人成绩单上，写着综合排名第一的文字。

    不过与此同时，我也知道了一项悲伤的事实：我明明花了比平时更多的时间在解题上，然而我放下笔的时间，却仍旧比班上任何人都还要早。

    早知如此，我之前正常考试就好了！

    我至今为止努力拼速度解题的辛劳到底算什么！

    不不不，冷静点。沉静下来仔细想想。正因为有竞速破关的那段日子，我的解题速度──思考速度才上升。所谓的进步就是更加效率化。那段日子并不是白费，一切都是努力的恩赐，绝对不是什么宛如赛之河原堆石子那般徒劳无功。

    「老师，考得如何？老师这次考试的成绩每一科分数都很异常，相信第一名也──」

    这时候，社长探头瞄了瞄我的成绩单。「哇！哇喔！你终于办到了呢！真厉害、真厉害！」

    「对吧？我可是第一名喔！快夸我、快夸我！」

    没错，我可是很厉害的，以我的能力做得到。

    「哎呀，真的让我老实地觉得你很厉害！毕竟你有好几科是满分。虽然很不甘心，不过看到这个排名我也认了。这次换我请客啊……我的钱要被贪婪的败北女角给抢走了……」

    不小心听到我的排名，班上同学不禁「神宫寺同学考第一吗？」、「好厉害！」、「神宫寺同学头脑真聪明！」、「这下白崎的天下也结束了」地说着，一个个喧嚣吵闹地来称赞我。

    没错没错！多称赞一点！把我给捧上天！

    我搁笔的速度可是比你们几个还早，还是考到了这种成绩喔！

    再多称赞我一点吧！我可是拥有看穿正确解答之力──燃犀之见的人！

    不过话说回来，大家应该没有听到「贪婪的败北女角」这句话吧？我唯独担心这一点。

    「别这样啦～我会很不好意思。」我夸张地刻意摆了摆手。

    接着，我与传送冷漠视线的德雷克和多米诺乐团──浅野丽良对上了视线。

    她那个「又来了」的神情是怎样？有什么意见吗？我想做什么是我的自由吧？有什么关系？又没给你添麻烦！别用那种带有怜悯意味的眼神看我。

translat to Japanese :喜びの中で.私はついにこの喜びを細かく味わうこともできる.何度か耐えられないのかわからない. 前日の憂慮を心配し,後日の喜びを喜びさせる.マッキヴィリーが言ったように,人間の脳は三種類に分かれている.一は,自己能力をよく理解する.二は,他人を理解する.三は,自分の能力を理解することにも理解することにもならない.一は非常に優れている.二はかなり優れている.三はかなり無力である.さて言うまでもなく,私は第一の脳である".その証拠は,私はいつもみんなを簡単に下ろしている.一は,個人成績を定期的にチェックし,総合的な文字をランク付けする.それと同時に,私は異論的なことを知っていました.私は第一の部分に,自分の能力を理解することにもならない.私は通常の時間よりも多くの時間を費やした結果の解明に費やした.しかし,私の功績の課題は,非常に高い,そして第三の段階では,かなり無力です.それ以外,私は,すべての時間を費やした結果の解明に費やした.しかし,私の課題は,古い日々の課題に,どんな速度も,どんなものでも,どんなものでも,どんなものでも,どんなものでも,どんなものでも,どんなものでも,どんなものでも,どんなものでも,どんなものでも,どんなものでも,どんな速度,どんなものでも,どんなものでも,どんなものでも,どんなものでも,どんなもの,どんなものでも,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,どんなもの,もの,どんなもの,もの,どんなもの,どんなもの,もの,どんなもの,どんなもの,もの,どんなもの,もの,または,どんなもの,どんなもの,どんなもの,もの,または,どんなもの,どんなもの,どんなもの,もの,もの,どんなもの,もの,もの,または,もの,または,どんなもの,もの,どんなもの,または,どんなもの,もの,または,もの,もの,または,もの,もの,どんなもの,または,または,もの,もの,または,もの,または,どんなもの,もの,または,もの,もの,もの,または,または,または,または,どんなもの,または,もの,または,もの,または,または,もの,もの,もの,または,または,もの,または,または,または,または,または,もの,または,または,もの,もの,または,または,または,または,もの,または,または,または,または,または,または,または,もの,または,または,または,または,または,もの,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または,または, 校長は,私の成績をチェックした. "うー!うー!あなたはついにこのランキングを入れました!本当に強烈,本当に強烈!""うー!私は第一の!快速,快速,誇らしげ!!""うー,私は本当に強烈,私の能力でそれをできる. "うー,本当に私はあなたが本当に強烈な感じさせました!何と多くの科目が満員です. 喜んでこのランキングを見るにもかかわらず,私はこのランキングも認めました. "この度"""私は"お客様...貪欲な敗戦は北角に奪われた...私のランキングを無注意に聞いて,班は"宮廷の学問を禁止するのですか?"と同学団の"強烈な角"を担う. "宮廷と神宮の無知の見地! 賢明な見地"この下にある女士は, "天下は,私の上に燃え,私の上にある!"と,私の意見は,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,私の意見に,です. 自由とは何か?それには何の関係があるか.また,あなたに悩まさない.わたしを憐れみ深い目で見てはならない.

2.translat to English :
I can finally taste this pleasure, and I don't know how long I've been able to bear it. I'm worried about the past and happy about the future. MacKivilly said that people's brains are divided into three types. The first is that they know their own abilities; the second is that they understand others; the third is that they don't understand their abilities or understand the strength of others. The first is extremely good, the second is quite excellent, and the third is quite inadequate - no need to say I'm the first type of brain". The proof is that I'm very easy on everyone. The first is that I can regularly check the results of the individual results, with a number of words that are summarized. "Oh, I'm really strong, and I can do it with my ability". "Oh, I really feel that you are really strong! After all, you have a lot of subjects. Although very unhappy, I also recognize this list. "This time", "O my guest". "The money is going to be robbed by the greedy North Pole". "I'm not careful to hear my ranking, the class is going to be uninterrupted with the "Do you have the best education?" "Oh, I really feel that you have the best education!" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education?" "Do you have the best education". "Do you have the best education". "Do you have the best education". "Do you have the best education". "Do you have the best education". "Do you have the best education". "Do you have the best education in the world, and you have the best education in the world". "Do you have the best education in the world". "Do you have the best education in the world". What does my freedom mean? What does it matter? I have not troubled you. Do not look at me with a pitiful eye.
